### ISSUE TITLE
Asciidoc xref links: WikiLinkOpen doesn't open links without hash

### ISSUE THREAD

**Issue by mgoral** [lervag/wiki.vim#135]
Hi!

According to the help file, wiki.vim now supports these two forms of Asciidoc's inline macros: 

```
xref:anchor[]
xref:file.adoc#anchor[]
```

However, I wonder if it should also support the first form (without the hash), but leading to another file instead of the anchor in current file:

```
xref:file.adoc[]
```

It looks like it is correct Asciidoc's document-to-document cross reference: https://docs.asciidoctor.org/asciidoc/latest/macros/inter-document-xref/#navigating-between-source-files

Currently, if user creates such xref by hand (because WikiLinkOpen automatically creates a version with hash), wiki.vim will do nothing and will print "Can't open link before you've saved the current buffer", which is also a different behaviour than if we had the version with hash (which would allow following the link without saving current buffer).

**Comment by lervag** [comment]
First, sorry for the extremely slow response!

I've pushed an update that I think fixes a couple of minor issues with adoc links. I also allowed this form as you suggest by matching against the `.adoc` at the end of the url string. So, for `xref:string[]`, the `string` will in general be recognized as an anchor pattern, except with `string ~= ".adoc$"`. I believe this should behave as expected in most cases?

> ...wiki.vim will do nothing and will print "Can't open link before you've saved the current buffer", which is also a different behaviour than if we had the version with hash (which would allow following the link without saving current buffer).

I strongly suggest that you use the `set hidden` option in Vim (and read the corresponding docs). Learn to navigate between buffers if you have not already done so. It is a very useful feature!
